src/NServiceBus.AcceptanceTests/Core/Diagnostics/When_endpoint_is_started.cs

```diff
@@ -0,0 +1,53 @@
+namespace NServiceBus.AcceptanceTests.Core.Diagnostics
+{
+    using System.IO;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+    using AcceptanceTesting.Customization;
+
+    public class When_endpoint_is_started : NServiceBusAcceptanceTest
+    {
+        static string basePath = Path.Combine(TestContext.CurrentContext.TestDirectory, TestContext.CurrentContext.Test.ID);
+        
+        [Test]
+        public async Task Should_add_licensing_section_to_diagnostic_file()
+        {
+            // TestContext.CurrentContext.Test.ID is stable across test runs,
+            // therefore we need to clear existing diagnostics file to avoid asserting on a stale file
+            if (Directory.Exists(basePath))
+            {
+                Directory.Delete(basePath, true);
+            }
+            
+            await Scenario.Define<Context>()
+                .WithEndpoint<EndpointThatWithDiagnosticsEnabled>()
+                .Done(c => c.EndpointsStarted)
+                .Run()
+                .ConfigureAwait(false);
+            
+            var endpointName = Conventions.EndpointNamingConvention(typeof(EndpointThatWithDiagnosticsEnabled));
+            var startupDiagnosticsFileName = $"{endpointName}-configuration.txt";
+
+            var pathToFile = Path.Combine(basePath, startupDiagnosticsFileName);
+            Assert.True(File.Exists(pathToFile));
+            
+            var diagnosticContent = File.ReadAllText(pathToFile);
+            Assert.True(diagnosticContent.Contains("\"Licensing\""));
```

Comment:
Is it worth parsing the doc, extracting the Licensing node and running an Approval Test on it? To do that properly, you'd probably need to explicitly add a license to the endpoint in this test.

Reply:
I did think about that. The drawback is that if the license is valid it can be used (as the source is public). Also, once it expires the test would fails with no obvious reason. An expired license would work for this purpose I think.

Reply:
Good point. An expired license won't be selected I think. Maybe it's best to leave it like this and not make any assertions at this point. Especially if we add "other license" details, we can't guarantee that the test will be consistent across machines.

Reply:
so I'll resolve this comment, since we think this is probably good enough.